app/waf/src/models/as3.model.ts

```diff
@@ -0,0 +1,238 @@
+import {Adc, Application, Service, Pool, Member, Wafpolicy} from '.';
+
+export class AS3Object extends Object {
+  class: string;
+
+  constructor() {
+    super();
+  }
+}
+
+export class AS3Pointer extends Object {
+  use: string;
+
+  constructor(name: string) {
+    super();
+
+    this.use = name;
+  }
+}
+
+export class AS3JSONObject extends Object {
+  [key: string]: Object;
+}
+
+export class AS3DeployRequest extends AS3Object {
+  action: string;
+  targetHost: string;
+  targetPort: number;
+  targetUsername: string;
+  targetPassphrase: string;
+  declaration: AS3Declaration;
+
+  constructor(params: {[key: string]: Object}) {
+    super();
+
+    let adc = <Adc>params['adc'];
+
+    this.class = 'AS3';
+    this.action = 'deploy';
+    this.targetHost = adc.host;
+    this.targetPort = adc.port;
+    this.targetUsername = adc.username;
+    this.targetPassphrase = adc.passphrase;
+    this.declaration = new AS3Declaration(params);
+  }
+}
+
+export class AS3Declaration extends AS3Object {
+  schemaVersion: string;
+  id: string;
+  tenants: AS3Tenant[];
```

Comment:
Why not use `tenants: Object` directly, then inline `toJSON()` function in constructor. 

So that we can leverage `JSON.stringify(<AS3 series class type>)` to convert them to string, and `JSON.parse(<json object string>)` to convert string to json object. 
The Object <-> Json process becomes uniform.

Reply:
The structure of AS3 JSON declaration is not consistent with object structure. Many objects, like pool, waf and etc, need to be declared in the same level with service. So we need construct the JSON structure by ourselves.

Reply:
got it.